gh-117999: use generic algorithm in complex_pow() if base has special components (#123283)
python/cpython: Misc/NEWS.d/next/Library/2024-08-24-10-46-35.gh-issue-117999.5K_BiA.rst

> Use single algorithm for complex power (case of small integer exponents was
> handled before by a special method).  Patch by Sergey B Kirpichev.

Reviewer: ```suggestion
Use a single algorithm for complex exponentiation (the case where the exponent
is a small integer was previously handled separately).  Patch by Sergey B Kirpichev.
```
